src/main/java/com/simibubi/create/content/trains/entity/Train.java

```diff
@@ -662,69 +728,59 @@ public Pair<Train, Vec3> findCollidingTrain(Level level, Vec3 start, Vec3 end, R
 			if (train.graph != null && train.graph != graph)
 				continue;
 
-			Vec3 lastPoint = null;
-
-			for (Carriage otherCarriage : train.carriages) {
-				for (boolean betweenBits : Iterate.trueAndFalse) {
-					if (betweenBits && lastPoint == null)
-						continue;
+			// Compute cache for trains that don't have it yet
+			if (train.collisionCache == null || !train.collisionCache.isValid()) {
+				train.updateCollisionCache();
```

Comment:
Instead of doing this lazily is there an easy way to do it for all Trains ahead of time somewhere in this method?
https://github.com/Creators-of-Create/Create/blob/f741b5b86e6e474fcb0de0aeb9286565f9b73c6c/src/main/java/com/simibubi/create/content/trains/GlobalRailwayManager.java#L219